### ISSUE TITLE
Install AROS on CF Card HowTo?

### ISSUE THREAD

**Issue by hgw77** [ApolloTeam-dev/AROS#15]
I tried to follow the decription for `Building Your Own Mini AROS Distribution` 

1. Build was successfull done on my linux box 
2. I packed `bin/amiga-68k` into tar and copied that to my windows workstation 
3. unpacked and copied the content on my Vampire-SD-Card.  
4. put that sd-card into my Vampire and boot my Amiga OS 3.9 System
5. copied the AROS content from my SD on my AROS Boot Partition and made this Partition Bootable
6. Copied vcontrol into the C Folder of my AROS Boot partition
7. tried to adjust the startup-sequence but there is no `boot/amiga/AROSBootstrap ROM boot/amiga/aros.hunk.gz` so I added on the beginning of the startup-sequence `vcontrol mr=boot/amiga/aros.rom`
9. reboot and my amiga is hanging

My Destination Amiga is an A500+ (2mb Chipmem) with Vampire Core 2.12 and Indivision V2

I am not sure maybe I missed something but I have no clue whats wrong here. Maybe a step by stepdocu howto build my own AROS from source to Boot on my Amiga would be helpfull

**Comment by Mingo12** [comment]
I have exactly the same problem:

https://apollo-core.com/knowledge.php?b=10&note=31640&z=gEems3

**Comment by hgw77** [comment]
OK I misssed the step to copy the `aros.rom` from the root of the AROS repo into `bin/amiga68k/boot/amiga/` now after `vcontrol mr=boot/amiga/aros.rom` my Amiga ends up in a grey/black screen where the system is telling me 

```
Software Failure
Task: 0x080111148 - Exec Bootstrap Task
Error: 0x80000008 - Privilege violation error
PC: 0x00E3F154
Wait for disk activity finish
```

![BootError](https://user-images.githubusercontent.com/5598283/90982201-e2568500-e565-11ea-8ecf-6b8e9efbea84.jpg)

By the way it is always the same Memory Address 0x080111148 and  0x80000008

For me it looks like the AROS rom crashes at boot :-(
Booting AROS from DH1 with the highest Bootprio (on DH0 I have my OS3 System), but I guess that doesnt matter.....

**Comment by ronybeck** [comment]
Does ApolloOS (pre-built) work on the same system?

**Comment by ronybeck** [comment]
So I updated the startup sequence and also the screenmode prefs so that one should be able to boot.  You will need to add a line at the beginning though:

vcontrol mr=boot/amiga/aros.rom

This means that you will need to copy vcontrol into c: and aros.rom into sys:boot/amiga/aros.rom (but you knew that already).

If this doesnt work, pleas update this ticket.  I will try this sometime soon in parallel.

**Comment by hgw77** [comment]
> Does ApolloOS (pre-built) work on the same system?

No, I tried both (pre-build and self-build). 

Note: Because I have no 16GB CF card left I wrote the prebuild ApolloOS image to a 16GB SD Card. Than I mounted the SDCard on WinUAE and copied the Data (on DH0 Partition) to my Amiga CF Card DH1 Partition and booted from their.

By the way, it would be nice to have a prebuild CF Card Image that contains only the AROS system, just for testing? 
The current ApolloOS Image with a size of 4.3GB can only fit on a 16GB CF Card. That is just for testing a little bit to heavy. An Image only with prebuild AROS should not bigger than 100MB (I guess) ;-)

**Comment by ronybeck** [comment]
The boot process for AROS a normal amiga is like this:

* Kickstart boots from boot drive (e.g. DH0) and runs s/startup-sequence
* Startup-sequence executes "vcontrol mr=boot/amiga/aros.rom" and resets
* aros will boot from boot drive

You need to intercept both "boots" and select boot device (DH1 in your case).  Did you do this?
Also, do you use vcontrol to speed up IDE?  I would turn this off temporarily until we know why it doesn't boot.  The error message you posted looks to be right at the begining.  Before it has even executed the startup-sequence.  And a privilege violation error here doesn't make much sense to me.  Either AROS can't read your disk properly or I am wrong about where in the boot sequence this error happens.  This starts to sound like disk/cable problems (but may not be).

My goal is to have a minimalistic aros like you say.  But for you there is little benefit because you anyway just copy files to another drive.  My image would contain the same files.  I would take the newer startup-sequence I committed last week and uncomment the "echo" lines.  It prints some debug throughout the boot.  Maybe (if I am wrong about where it fails) you will see where in the boot it fails or if it is even booting at all.

**Comment by Mingo12** [comment]
I also have tested the pre build Version (ApolloOS_16GbF3C_2208) but there is no difference. I decided to compile it again, since there was made some modifications on the code, but also this time i had no success. On the RGB Output is no image and on HDMI i see only the Vampiere boot Screen. I have the core 2.11 installed, because version 2.12, from what i understand, have no Vampire IDE Support. Is the 2.12 core required to run AROS 68K?

**Comment by ronybeck** [comment]
> I have the core 2.11 installed, because version 2.12, from what i understand, have no Vampire IDE Support.

No.  This is just wrong.  Maybe you are thinking of the V1200 core where the onboard IDE isn't yet supported?

**Comment by Mingo12** [comment]
Maybe i misunderstood something, but I've read several times that Core 2.12 doesn't support the Vampire IDE, and there should be two versions, one with Vampire IDE (which I can't find) and the other with A1200 IDE support. Sorry but this is a bit confusing for me... So I flashed the 2.12 core from the download section and as expected the Amiga no longer boots from the Vampire IDE, but only from the A1200 IDE. But that's not the problem. I changed the startup sequence of ApolloOS like the advice of a Forum-Member:

  Do you ask "Load Aros Rom?"
   When WARN
     VControl MR = boot / amiga / aros.rom
   EndIF

And with this modification and Core 2.12 I got one step further. Now I see a request window on the RGB output. On the HDMI i see nothing.

![aros](https://user-images.githubusercontent.com/40700774/92960681-55209500-f46e-11ea-8484-c3e090c73673.jpg)

If I click on "Retry" nothing happens, and if I click on "Cancel" the Amiga loads something and then restarts.

**Comment by ronybeck** [comment]
@Mingo12 Hey!  That is a good step forward.  If you close the shell in the background, then click "retry", you will probably find it boots.

**Comment by ronybeck** [comment]
So I looked into this problem a bit more.  But let me first define what I mean because there are a couple of different problems listed here.

**On first boot of AROS the Amiga hangs**
This can be reproduced and is due to a bug in recent versions of vcontrol.  Vcontrol should check that it hasn't already mapped the AROS rom before trying again fails and so the Amiga is continuously mapping the rom and rebooting it.  The solution for now is to use vcontrol from earlier versions, such as 1.5.

There is a report of another problem where AROS just doesn't start at all.  I cannot reproduce this.  I have 5 vampirised amigas (A500+, A600, A2000, A1200 and V4).  None of them show these problems.

For completeness, I am uploading the contents of my Aros boot drive for other people to try.

[Download the Zip](https://drive.google.com/file/d/1SNcCQ2ZHzTUJ-nNOL7IlsruUmBmeE8r8/view?usp=sharing)

I would ask that people try this and if this fails to boot, that you open a new ticket.  In that ticket can you please list:
* Amiga Model and Revision
* Vampire Model (  [click here to identify yours ](https://www.majsta.com/modules.php?name=News&file=categories&op=newindex&catid=11) )
* your hard disk setup including disk type, cable length and if you are using any adapters
* any expansions you have installed
* Anything else you think is unique about your amiga

Reproducing the problem is crucial and right now I don't know how to reproduce this problem and I need more information.

**Comment by hgw77** [comment]
hey @ronybeck sorry for the delay :-)

1. downloaded the zip
2. copied it to my sd card on my PC 
3. copied it from sd card to my second amiga partition (FFS filesystem) on my Amiga
4. reboot amiga and boot from second partition
5. crash :-/

so there is something different with my Amiga500+ REV8a

**Comment by ronybeck** [comment]
What does "crash" mean exactly?
Is it EXACTLY the same as your screen shot in the second post?

I don't consider this to be a bug because the problem isn't reported by anyone else.  I suspect you have problems with your CF card or cable.  Perhaps you could ask in the chat for guidance.  I don't think any code changes I can make will fix your problem.

**Comment by hgw77** [comment]
sorry the crash that I mean is exactly the same that I posted above

Software Failure
Task: 0x080111148 - Exec Bootstrap Task
Error: 0x80000008 - Privilege violation error
PC: 0x00E3F154
Wait for disk activity finish

**Comment by hgw77** [comment]
OK after a try with my second Amiga500 and the same result I tried another CF-card (Scandisk Ultra 16GB) that I found on my desk and oh wonder with this Card it was working. So it seems my CF-card was not compatible. This CF-card (TRS 4GB industrial grade) was working flawlessly over years with my OS3.1 installation so I was thinking that could not be the culprit but ist was ;-)
